### ISSUE TITLE
Support import TCX file

### ISSUE THREAD

**Issue by juanro49** [SamR1/FitTrackee#529]
Hello!

I have set up an instance of FitTrackee to test the software a bit, and I have seen that at the moment we can only import files in GPX format. Could you add support for the TCX format which is also very popular, supporting much more information than GPX and used as export format for different wearables like Garmin?

For example, Amazfish use TCX as output file (and I suggested it to support FitTrackee) https://github.com/piggz/harbour-amazfish/issues/362

**Comment by SamR1** [comment]
Hi,

For now, only the gpx format is supported (but [3rd-party tools](https://samr1.github.io/FitTrackee/en/third_party_tools.html#import-tools) can be used to add files in other formats or from other sources).
Support for other formats is planned, but ideally requires a rewrite of workout file processing (see #194).

**Comment by juanro49** [comment]
Thank you for considering it for the future. I will try the Strava-to-FitTrackee tool to import my activities from Strava. FitTrackee has a huge job and it's a platform that's very necessary!

**Comment by slackline** [comment]
You could give the [tcx2gpx](https://pypi.org/project/tcx2gpx/) package a whirl to batch convert TCX to GPX files. If it doesn't work please [let me know](https://gitlab.com/nshephard/tcx2gpx/-/issues).

In time I'll see if the functionality can be incorporated into third party tools.

**Comment by slackline** [comment]
Thanks for giving `tcx2gpx` a whirl @juanro49 .


I've created an [issue on the tcx2gpx repository](https://gitlab.com/nshephard/tcx2gpx/-/issues/19) to track work on this. If you're able to run `tcx2gpx -j 1` this will force a single core to be used and help narrow down the offending `.tcx` file which I would be grateful if you could share with me if you don't mind as I can then investigate what is causing the problem.

If you are able to share the file on the above linked issue that would be great, or message me directly with it to avoid this issue going off track.

Cheers.

**Comment by juanro49** [comment]
> Thanks for giving `tcx2gpx` a whirl @juanro49 .
> 
> I've created an [issue on the tcx2gpx repository](https://gitlab.com/nshephard/tcx2gpx/-/issues/19) to track work on this. If you're able to run `tcx2gpx -j 1` this will force a single core to be used and help narrow down the offending `.tcx` file which I would be grateful if you could share with me if you don't mind as I can then investigate what is causing the problem.
> 
> If you are able to share the file on the above linked issue that would be great, or message me directly with it to avoid this issue going off track.
> 
> Cheers.

with -j 1, I have same exact error. This is the `.tcx` file [Carrera_de_tarde.txt](https://github.com/user-attachments/files/15842304/Carrera_de_tarde.txt)

**Comment by SamR1** [comment]
Hi @juanro49 

I have started to implement TCX support.
I get an error when I upload the `.tcx` file above, due to `AltitudeMeters` values.

`AltitudeMeters` value is in meters, but the values in the file don't seem to correspond to meters.

For example:
```
<Trackpoint>
  <Time>2024-04-18T18:50:54Z</Time>
  <Position>
    <LatitudeDegrees>39.49898</LatitudeDegrees>
    <LongitudeDegrees>-6.36318</LongitudeDegrees>
  </Position>
  <AltitudeMeters>-40000</AltitudeMeters>
</Trackpoint>
```

According to Open-Elevation API, the altitude of this point is 347m (https://api.open-elevation.com/api/v1/lookup?locations=39.49898,-6.36318)

Do you know if this is related to some device settings? Thank.


cc @slackline

**Comment by SamR1** [comment]
Thanks!
One solution would be to correct elevation (cf. https://github.com/SamR1/FitTrackee/issues/701), but this is not yet planned.

Are the files currently added to FitTrackee from Strava? Are elevations fixed in `.gpx` files?

**Comment by juanro49** [comment]
> Thanks! One solution would be to correct elevation (cf. [#701](https://github.com/SamR1/FitTrackee/issues/701)), but this is not yet planned.
> 
> Are the files currently added to FitTrackee from Strava? Are elevations fixed in `.gpx` files?

Yes, if I download from Strava the TCX uploaded file in GPX format, it is already downloaded with the elevation fixed.

**Comment by slackline** [comment]
Could it be that some app/devices record unknown elevation as a non-sensical value?

If so as a, general rule, would it be reasonable to convert such values to missing when importing?

Looks like the lowest point on land below sea-level is [-153m](https://en.wikipedia.org/wiki/List_of_places_on_land_with_elevations_below_sea_level) and so handling anything with `elevation < -200` as something that should be converted to missing might allow FitTrackee to continue processing the import.

**Comment by SamR1** [comment]
released in v0.10.0b1
